Restructure dashboard controls docs and present controls as panels (#5664)
elastic/docs-content: explore-analyze/visualize/add-controls.md

---
mapped_pages:
  - https://www.elastic.co/guide/en/kibana/current/add-controls.html
navigation_title: Add controls
type: how-to
description: Add interactive filter controls to your Kibana dashboards to help users explore data with options lists and range sliders.
applies_to:
  stack: ga
  serverless: ga
products:
  - id: kibana
---

# Add Options list and Range slider controls to dashboards [add-controls]

> Add interactive filter [controls](dashboard-controls.md) to your dashboards to help viewers explore data without writing queries. This page covers how to add, edit, and remove Options list and Range slider controls.

Reviewer: ```suggestion
Add interactive [controls](dashboard-controls.md) to your dashboards to help viewers filter data without writing queries. This page covers how to add, edit, and remove Options list and Range slider controls.
```

Author: Added in 3a54a7cec

* An existing dashboard open in **Edit** mode
* A [data view](../find-and-organize/data-views.md) configured with fields available for filtering

:::{include} ../_snippets/control-limits.md
:::

## Add Options list and Range slider controls [create-and-add-options-list-and-range-slider-controls]

To add interactive Options list and Range slider controls, create the controls, then add them to your dashboard.

1. Open or create a dashboard.
2. Open the **Create control** flyout:

    - {applies_to}`serverless: ga` {applies_to}`stack: ga 9.2` In **Edit** mode, select **Add** → **Controls** → **Control**.
    - {applies_to}`stack: ga 9.0-9.1` In **Edit** mode, select **Controls** → **Add control** in the dashboard toolbar.

3. Choose how to populate the values available in the control:

    - **Select a field**: base the control on a [data view](../find-and-organize/data-views.md) field. The control offers the values found in that field.

        1. From the **Data view** dropdown, select the data view that contains the field you want to use.
        2. In the **Field** list, select the field you want to filter on.

>     - {applies_to}`serverless: ga` {applies_to}`stack: ga 9.5` **Write a query**: populate the control with the results of an {{esql}} query. Use this for high-cardinality fields, or when you want to filter or otherwise shape the values the control offers.

Reviewer 2: Why for high-cardinality fields? Do we consider it will be more performant? If so, we should add that reason here.

Author: Added in 3a54a7cec

        1. Write an {{esql}} query that returns a single column. The column determines the field the control filters on and the values it offers. Use a command such as `STATS BY` to return a single column.
        2. Run the query to preview the values it returns under **Values preview**. If the query returns more than one column, select a column or narrow the query. If it returns no values, edit the query and run it again.

        :::{tip}
        Because the values come from a query, you can also chain the control to a [variable control](add-variable-controls.md) by referencing its variable with the `?variable_name` syntax.
        :::

> 4. Under **Control type**, select **Options list** or **Range slider**. Range sliders are only compatible with numeric fields. When you populate a Range slider with a query, the query results set the slider's minimum and maximum values.

Reviewer 3: > When you populate a Range slider with a query, the query results set the slider's minimum and maximum values.

This is true when you select a field directly, too (without using the query builder). Perhaps we rephrase this?

Author: Added in 3a54a7cec

6. Select **Save** to add the control to the dashboard. The control appears right away in **Edit** mode, where you can test it and adjust its settings before viewers see it.
7. {applies_to}`serverless: ga` {applies_to}`stack: ga 9.4` Choose where the control appears. New controls are pinned to the top of the dashboard by default, where they apply to all panels. To move a control into the dashboard body, hover over the control and select **Unpin**. To move it back, select **Pin to dashboard**. Once unpinned, you can move, resize, and arrange the control on the dashboard like any other panel, including placing it inside a [collapsible section](../dashboards/arrange-panels.md#collapsible-sections) to scope it to that section's panels. Placement changes which panels a control filters. For details, refer to [How controls affect the dashboard](dashboard-controls.md#controls-scope).
8. Save the dashboard. The control becomes a permanent part of it, and viewers can use it to filter the relevant panels.

## Manage Options list and Range slider controls [manage-controls]

> After a control is on your dashboard, you can change its settings, reset its selections, move it, adjust its width, or delete it. How you access these actions depends on whether the control is pinned:

Reviewer: ```suggestion
After adding a control to your dashboard, you can change its settings, clear its values, move it, adjust its width, or delete it. How you access these actions depends on whether the control is pinned:
```

Author: Added in 3a54a7cec

- **When pinned**, hover over the control to reveal its action icons.
- {applies_to}`serverless: ga` {applies_to}`stack: ga 9.4` **When unpinned**, hover over the control to reveal its action icons, or open its panel menu, which adds standard panel actions such as **Duplicate** and **Copy to dashboard**.

| Action | Description |
| --- | --- |

> | **Clear** | Reset the control's selections without changing its settings. Available only when the control has a selection. |

Reviewer: ```suggestion
| **Clear** | Clear the selected values without changing the control's settings. Available only when a value is selected. |
```

Author: Added in 3a54a7cec

| **Unpin** or **Pin to Dashboard** {applies_to}`serverless: ga` {applies_to}`stack: ga 9.4` | Move the control between the top of the dashboard and the dashboard body. For details, refer to [Pinned and unpinned controls](dashboard-controls.md#pinned-unpinned-controls). |

> | **Edit** | Change the control's settings in the **Edit control** flyout. For the full list, refer to [Dashboard control settings](dashboard-control-settings.md). |

Reviewer 3: You can do more than just edit settings - you can change the field, control type, query, etc. Not sure if we need to use a broader term here?

Author: Added in 3a54a7cec
